universe/ValueRefs.h

```diff
@@ -377,6 +424,15 @@ struct FO_COMMON_API Operation final : public ValueRef<T>
 
     unsigned int GetCheckSum() const override;
 
+    std::unique_ptr<ValueRef<T>> Clone() const override {
+        std::vector<std::unique_ptr<Operation<T>>> operands(m_operands.size());
+        for (const auto& operand : m_operands) {
+            operands.emplace_back(operand ? std::move(std::make_unique<ValueRef<T>>(operand->Clone())) : nullptr);
+        }
+        auto retval = std::make_unique<Operation<T>>(m_op_type, std::move(operands));
+        return std::move(retval);
```

Comment:
probably should not `std::move` to return; it prevents copy elision.

Reply:
Fixed.